src/pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -75,8 +76,7 @@ def resume_start(self, checkpoint_path: Optional[_PATH] = None) -> None:
         3. from `checkpoint_path` file if provided
         4. don't restore
         """
-        self.resume_checkpoint_path = self._hpc_resume_path or checkpoint_path
```

Comment:
Okay, so this is the breaking change you mention in the changelog. No longer do we auto detect the file, and only look at if if ckpt_path="hpc". LGTM

Reply:
Would you mind adjusting the docstring of the first item above: "1. from HPC weights if found" -> "1. from HPC if checkpont_path has the special name 'hpc'."